src/main/java/org/grouplens/grapht/Instantiators.java

```diff
@@ -140,13 +146,20 @@ public ProviderInstantiator(Instantiator prov) {
         @Override
         public Object instantiate() throws ConstructionException {
             Provider<?> provider = (Provider) providerInstantiator.instantiate();
+            LogContext mdcContextProvider = LogContext.create();
+            logger.trace("invoked provider",provider);
             try {
+                mdcContextProvider.put("org.grouplens.grapht.currentProvide", provider.toString());
                 return provider.get();
-            } catch (Throwable th) {
+            }
+            catch (Throwable th) {
```

Comment:
Style point: could you move the `catch` (and subsequent `finally`) on to the same lines with their braces?